test/server/cards/03_TWI/leaders/NuteGunrayVindictiveViceroy.spec.ts

```diff
@@ -0,0 +1,58 @@
+describe('Nute Gunray, Vindictive Viceroy', function () {
+    integration(function (contextRef) {
+        describe('Nute Gunray\'s leader undeployed ability', function () {
+            beforeEach(function () {
+                contextRef.setupTest({
+                    phase: 'action',
+                    player1: {
+                        groundArena: ['battlefield-marine'],
+                        spaceArena: ['green-squadron-awing'],
+                        leader: 'nute-gunray#vindictive-viceroy',
+                        resources: 4,
+                    },
+                    player2: {
+                        hand: ['admiral-yularen#advising-caution', 'superlaser-blast'],
+                    },
+                });
+            });
+
+            it('should create a battledroid token if two or more friendly units defeated this phase', function () {
+                const { context } = contextRef;
+
+                context.player1.passAction();
+
+                // Defeat units
```

Comment:
We need a few more test cases here to confirm the trigger is working correctly (can move between them with `.moveToNextActionPhase()`.

Phase 1: defeat 2 enemy units and only 1 friendly unit, activate his ability. nothing should happen.
Phase 2: defeat 2 friendly units, confirm that his ability triggers.
Phase 3: defeat 0 units and activate his ability, nothing should happen. this confirms that it isn't counting units defeated in the previous phase